ocaml/parsing/jane_syntax.ml

```diff
@@ -92,36 +92,53 @@ module Comprehensions = struct
      v}
   *)
 
-  let comprehension_expr names x = Expression.make_jane_syntax feature names x
+  (* Call [Embedded_name.of_feature] once per list we use to increase sharing --
+     that way, the strings that Jane Syntax embeds in the parsetree (in
+     attribute and extension node names) are shared.
+  *)
+  module Names = struct
+    type t = Embedded_name.t
+    let create : string list -> t = Embedded_name.of_feature feature
+
+    let for_range_upto : t = create [ "for"; "range"; "upto" ]
```

Comment:
I was expecting to see each of these names used twice: once for creation and once for checking. But the checking is still hard-coded. Do you think it would be an improvement to use these in checking, too?